bonobo/nodes/basics.py

```diff
@@ -314,6 +316,32 @@ def _Format(context, bag):
     return _Format
 
 
+@transformation_factory
+def MapFields(callFunc):
+    # TODO: Figure out how to more intelligently handle correctable errors
+    # Use this wrapper to call the function that we pass in and handle the error
+    # For now, handle the error by returning the unmodified value
+    def _wrapfunc(func, val):
+        try:
+            retval = func(val)  # Attempt to use the function
+        except:
+            retval = val  # If we get any error, return the value unmodified
+
+        return retval
+
+    @use_raw_input
+    def _MapFields(bag):
+        nonlocal callFunc
+
+        b = {key: _wrapfunc(callFunc, value) for key, value in bag._asdict().items()}  # iterate the wrapped callable
+        makenamedtuple = collections.namedtuple('Bag', ' '.join((b.keys())))  # constructor of the returned tuple
```

Comment:
You can use @use_context and get/set the input/output type from there. I think it's better than guessing the user would want this namedtuple.